typing/ctype.ml

```diff
@@ -2800,12 +2809,13 @@ and unify_row env row1 row2 =
         with Not_found -> ())
       r2
   end;
-  let fixed1 = row_fixed row1 and fixed2 = row_fixed row2 in
-  let more =
-    if fixed1 then rm1 else
-    if fixed2 then rm2 else
-    newty2 (min rm1.level rm2.level) (Tvar None) in
-  let fixed = fixed1 || fixed2
+  let fixed1 = fixed_explanation row1 and fixed2 = fixed_explanation row2 in
+  let more = match fixed1, fixed2 with
+    | Some _, _ -> rm1
+    | None, Some _ -> rm2
+    | None, None -> newty2 (min rm1.level rm2.level) (Tvar None)
+  in
+  let fixed = merge_fixed_explanation row1 row2
```

Comment:
That doesn't look right. Previously we were merging the result of the calls to `row_fixed`, here you're merging the `row_fixed` fields of `row1` and `row2`, these might be `None` while `fixed1` and `fixed2` might be `Some thing`.

Reply:
Indeed, I have fixed the error by making `merge_fixed_explanation` type more symmetric as `fixed option -> fixed option -> fixed option`.